test/unit/sessions.test.js

```diff
@@ -428,20 +428,28 @@ describe('Sessions - unit', function () {
       done();
     });
 
-    it('should remove sessions which have timed out on release', function (done) {
+    it('should remove all sessions which have timed out on release', function () {
```

Comment:
let's make sure that we test that old sessions are removed regardless of where they are in the list of sessions, the following four combinations test distinct looping behaviors that I think should cover it:
1) old, new [old sessions at the start]
2) new, old [old sessions at the end]
3) new, old, new [old sessions in the middle]
4) old, new, old [non-contiguous old sessions]